server/api.go

```diff
@@ -87,45 +89,44 @@ func (p *Plugin) handleJaaSSettings(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var jaasSettingsFromAction JaaSSettingsFromAction
-
 	bodyData, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		mlog.Debug("Unable to read request body", mlog.Err(err))
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	err1 := json.NewDecoder(bytes.NewReader(bodyData)).Decode(&jaasSettingsFromAction)
-	if err1 != nil {
+	err = json.NewDecoder(bytes.NewReader(bodyData)).Decode(&jaasSettingsFromAction)
+	if err != nil {
 		mlog.Debug("Unable to decode the request content as start meeting request or start meeting action")
 		http.Error(w, "Unable to decode your request", http.StatusBadRequest)
 		return
 	}
 
-	var user *model.User = nil
-	userID := r.Header.Get("Mattermost-User-Id")
+	var user *model.User
+	userID := r.Header.Get(mattermostUserID)
 	if userID != "" {
 		// Handle moderator
 		userResult, appErr := p.API.GetUser(userID)
 		if appErr != nil {
 			mlog.Debug("Unable to get the user", mlog.Err(appErr))
-			http.Error(w, "Forbidden", http.StatusForbidden)
+			http.Error(w, "Unable to get the user", http.StatusForbidden)
 			return
 		}
 		user = userResult
 	}
 
-	jaasSettings, err2 := p.getJaaSSettings(jaasSettingsFromAction.Jwt, jaasSettingsFromAction.Path, user)
-	if err2 != nil {
-		mlog.Error("Error getting JaaSSettings", mlog.Err(err2))
+	jaasSettings, err := p.getJaaSSettings(jaasSettingsFromAction.Jwt, jaasSettingsFromAction.Path, user)
+	if err != nil {
+		mlog.Error("Error getting JaaSSettings", mlog.Err(err))
 		http.Error(w, "Invalid JaaS settings", http.StatusBadRequest)
 		return
 	}
 
 	settingsJSON, err := json.Marshal(jaasSettings)
 	if err != nil {
 		mlog.Error("Error marshaling the JaaSSettings to json", mlog.Err(err))
-		http.Error(w, "Internal error", http.StatusInternalServerError)
+		http.Error(w, "Error marshaling the JaaSSettings to json", http.StatusInternalServerError)
```

Comment:
I think we should not send such error messages in the API response. Detailed errors should be printed in the server logs only.
```suggestion
		http.Error(w, "Internal server error", http.StatusInternalServerError)
```